File: produtos_flutter/lib/models/pagy.dart
class Pagy {
  int totalItems;
  int totalPages;
  int currentPage;
  int pageSize;

  Pagy({
    required this.totalItems,
    required this.totalPages,
    required this.currentPage,
    required this.pageSize,
  });

  factory Pagy.initial() {
    return Pagy(totalItems: 0, totalPages: 1, currentPage: 1, pageSize: 0);
  }

  factory Pagy.fromJson(Map<String, dynamic> map) {
    return Pagy(
      totalItems: map['totalItems'],
      totalPages: map['totalPages'],
      currentPage: map['currentPage'],
      pageSize: map['pageSize'],
    );
  }
}

File: ProdutosTeste/Models/Pagy.cs
public class Pagy<T>
{
    public int CurrentPage { get; set; }
    public int PageSize { get; set; }
    public int TotalItems { get; set; }
    public int TotalPages { get; set; }
    public List<T> Data { get; set; }

    public Pagy(List<T> data, int totalItems, int currentPage, int pageSize)
    {
        Data = data;
        TotalItems = totalItems;
        CurrentPage = currentPage;
        PageSize = pageSize;
        TotalPages = (int)Math.Ceiling(totalItems / (double)pageSize);
    }
}